src/kagglehub/handle.py

```diff
@@ -43,6 +43,28 @@ def to_url(self) -> str:
             return f"https://www.kaggle.com/models/{self.owner}/{self.model}/frameworks/{self.framework}/variations/{self.variation}"
 
 
+@dataclass
+class DatasetHandle:
+    owner: str
+    dataset_name: str
+    version: Optional[int] = None
+
+    def is_versioned(self) -> bool:
```

Comment:
nit: `has_version(self)`

Reply:
we use is_versioned to keep it consistent with the model handle